CONTRIBUTING.md

```diff
@@ -0,0 +1,73 @@
+# Contributing to Handshake
+
+## Code style
+
+This repository includes a file [.eslintrc.json](.eslintrc.json) which sets
+linting preferences for the project. The continuous integration bot will
+install [bslint](https://www.npmjs.com/package/bslint) which vendors a specific
+version of eslint. Before submitting a pull request, please make sure your code
+is clean of linting errors. If you choose to use bslint, it can be installed
+globally in your development environment:
+
+```
+npm install bslint -g
+cd hsd            # must be in repository root
+npm run lint      # command is defined in package.json
+```
+
+## Testing
+
+[bmocha](https://www.npmjs.com/package/bmocha) will be installed as a
+"developer dependency" if installed without the `--production` flag. The
+complete built-in testing suite can be run with the command:
+
+```
+cd hsd
+npm run test
+```
+
+You can specify a single test file to run with (for example):
+
+```
+cd hsd
+npm run test-file test/wallet-test.js
+```
+
+Before submitting a pull request, please make sure your code changes do not
+break any of the existing regression tests. We currently use GitHub Workflows
```

Comment:
... `any of the existing tests and linting rules`?

Reply:
Added. Also mentions not breaking lint rules above in the style section but hey. its docs. redundancy is key bc no one reads them anyway :-)